src/main/java/unrc/dose/UserProfileService.java

```diff
@@ -0,0 +1,42 @@
+package unrc.dose;
+
+import com.google.gson.Gson;
+
+/**
+* Class which calls user profile's methods and transforms they to json.
+*/
+public final class UserProfileService {
+  /**
+  *Invokes the method which creates a user profile and transforms to json.
+  **/
+  public static String userprofile(Integer userId, String displayName, String twitterId) {
+	  
+      return UserProfile.createUserProfile(userId, displayName, twitterId).toJson(true);
+
+  }
+  
+  public static String searchById (int userId) {
+	  if (!User.exists(userId)){
+          throw new UserNotFoundException(String.valueOf(userId));
+          }
+      	UserProfile up = UserProfile.findFirst("user_id = ?", userId); 
+      	return up.toJson(true, "user_id", "displayName","twitter_id");       	
+  }
```

Comment:
This method should return a `UserProfile`, a check to see if a profile exists for a particular userId should be either a precondition of this method or a part of this method's code.